pkg/env/env.go

```diff
@@ -0,0 +1,218 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package env
+
+import (
+	"context"
+	"log"
+	"testing"
+
+	"sigs.k8s.io/e2e-framework/pkg/conf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+	"sigs.k8s.io/e2e-framework/pkg/internal/types"
+)
+
+type Environment = types.Environment
+type Func = types.EnvFunc
+
+type actionRole uint8
+
+type testEnv struct {
+	cfg     *conf.Config
+	actions []action
+}
+
+// New creates a test environment with no config attached.
+func New() types.Environment {
+	return &testEnv{cfg: conf.New()}
+}
```

Comment:
This is probably me not knowing `internal` well enough, but should this be returning `Environment` instead?

Like, +1 to using `internal` to hide types you don't want users to see.  I guess it's not immediately clear to me what the user-visible API is though.

Reply:
This a good question. I did it for one reason initially, to avoid circular dependencies.
Then I started playing with type aliasing that points to internal types.  But this could be something that gets refined.